pandas/tests/extension/base/constructors.py

```diff
@@ -45,3 +46,14 @@ def test_series_given_mismatched_index_raises(self, data):
         msg = 'Length of passed values is 3, index implies 5'
         with tm.assert_raises_regex(ValueError, msg):
             pd.Series(data[:3], index=[0, 1, 2, 3, 4])
+
+    def test_from_dtype(self, data):
+        # construct from our dtype & string dtype
+        dtype = data.dtype
+
+        expected = pd.Series(data)
+        result = pd.Series(np.array(data), dtype=dtype)
```

Comment:
Do we have a restriction on what the result of `__array__` should be? Maybe not necessarily the scalars? 

So maybe better to do `list(data)` or `data.astype(object)`